src/components/segmentedControl/segmentedControl.svelte

```diff
@@ -80,63 +55,60 @@
     // a change event.
     if (!item) return
 
-    // Use data-is-interactive=true to prevent the menu from closing when selected. This implies
-    // there is interacitivity inside the menu item (e.g. a Toggle), which would be good for the user
-    // to see change state and allowing the user to manually close when ready. In other words, when we want
-    // the styles and menu navigation features of leo-menu-item, but we don't want the auto-close.
-
-    if (item.tagName === 'LEO-SEGMENTED-ITEM') {
-      selected = getValue(item)
+    value = getValue(item)
 
-      dispatch('select-item', {
-        value: selected
-      })
-    }
-
-    // When using keyboard navigation to call selectItem, ensure click handlers happen on the item,
-    // but not when this was called in a click handler, otherwise we'll get 2x clicks.
-    if (item.tagName === 'LEO-SEGMENTED-ITEM' && e.type !== 'click') {
-      item.click()
-    }
+    dispatch('change', { value })
   }
 
-  let pill: HTMLDivElement;
+  let pill: HTMLDivElement
 
   onMount(() => {
-    pill.addEventListener("transitionstart", () => {
-      segmentedControl.classList.add("transitioning");
-    });
-
-    pill.addEventListener("transitionend", () => {
-      segmentedControl.classList.remove("transitioning");
-    });
-  });
+    pill.addEventListener('transitionstart', () => {
+      segmentedControl.classList.add('transitioning')
+    })
+
+    pill.addEventListener('transitionend', () => {
+      segmentedControl.classList.remove('transitioning')
+    })
+  })
 </script>
 
 <div
   bind:this={segmentedControl}
-  class="leo-segmented-control"
-  class:isSmall={size === 'small'}
-  class:isTiny={size === 'tiny'}
+  class="leo-segmented-control size-{size}"
+  role="listbox"
+  tabindex="-1"
   on:keypress={(e) => {
     if (e.code !== 'Enter' && e.code !== 'Space') return
     selectItem(e)
   }}
   on:click={selectItem}
 >
-  <div class="pill" style:width={`${pillWidth}px`} style:left={`${pillPosition}px`} bind:this={pill} />
+  <div
+    class="pill"
+    style:width={`${pillWidth}px`}
+    style:left={`${pillPosition}px`}
+    bind:this={pill}
+  />
   <slot />
 </div>
 
 <style lang="scss">
+  :host {
+    position: relative;
```

Comment:
Is `inline-block` an appropriate display for this component (unfortunately all web components default to inline-block)?

Reply:
What? Oof. No, it should just be `block`.

Reply:
I've shot myself with this quite a few times :D

Reply:
Ooooh flex - Mr Fancy :laughing: